Simplify network controller unit test setup

Gudahtt opened the pull request:
## Explanation

The network controller unit tests have been refactored to simplify the setup steps related to the controller messenger. The messenger is now constructed by the `withController` helper function, rather than being manually setup in each test that references it.

Relates to https://github.com/MetaMask/core/issues/1197

## Manual Testing Steps

N/A

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

Gudahtt committed:
Simplify network controller unit test setup

The network controller unit tests have been refactored to simplify the
setup steps related to the controller messenger. The messenger is now
constructed by the `withController` helper function, rather than being
manually setup in each test that references it.

Relates to https://github.com/MetaMask/core/issues/1197

mcmire commented on app/scripts/controllers/network/network-controller.test.ts:
```diff
@@ -8239,18 +8063,26 @@ function mockCreateNetworkClient() {
+ * @param messenger - A controller messenger.
+ * @returns The controller options.
+ */
+function buildDefaultNetworkControllerOptions(
```
Hmm. I'm realizing now that the tests are a bit out of sync with respect to how the `withController` helper is written exactly. I like the way that we have it now in the `core` repo as it's simpler, so maybe we make it more consistent from this side? In other words what if `withController` were:

``` typescript
async function withController<ReturnValue>(
  ...args: WithControllerArgs<ReturnValue>
): Promise<ReturnValue> {
  const [{ ...rest }, fn] = args.length === 2 ? args : [{}, args[0]];
  const messenger = buildMessenger();
  const restrictedMessenger = buildNetworkControllerMessenger(messenger);
  const controller = new NetworkController({
    messenger: restrictedMessenger,
    trackMetaMetricsEvent: jest.fn(),
    infuraProjectId: 'infura-project-id',
    ...rest,
  });
  try {
    return await fn({ controller, messenger });
  } finally {
    await controller.destroy();
  }
}
```

Then we wouldn't need `buildDefaultNetworkControllerOptions` or `DEFAULT_INFURA_PROJECT_ID`. But in its place, we'd introduce `buildNetworkControllerMessenger`.

Gudahtt replied:
Good idea! I had noticed this difference, but thought it would be more work to resolve than it ended up being.

Done in https://github.com/MetaMask/metamask-extension/pull/18901/commits/3e464e7daa1af69770c1dc740e0e3d815fc6492b

Gudahtt committed:
Bring tests closer in alignment to core

The function to build default controller constructor parameters has
been removed, bringing these tests closer in alignment to the core
tests.

mcmire commented on app/scripts/controllers/network/network-controller.test.ts:
```diff
@@ -8301,7 +8111,10 @@ type WithControllerOptions = Partial<NetworkControllerOptions>;
   | [WithControllerCallback<ReturnValue>]
-  | [WithControllerOptions, WithControllerCallback<ReturnValue>];
+  | [
+      Omit<WithControllerOptions, 'messenger'>,
```
Ah sure, this makes sense 👍🏻

mcmire reviewed (approved):
Looks good!
